connaisseur/util.py

```diff
@@ -11,5 +13,77 @@ def normalize_delegation(delegation_role: str):
 
 def safe_path_func(callback: callable, base_dir: str, path: str, *args, **kwargs):
     if os.path.commonprefix((os.path.realpath(path), base_dir)) != base_dir:
-        raise InvalidFormatException("potential path traversal.", {"path": path})
+        msg = "Potential path traversal in {path}."
+        raise PathTraversalError(message=msg, path=path)
     return callback(path, *args, **kwargs)
+
+
+def get_admission_review(
+    uid: str,
+    allowed: bool,
+    patch: list = None,
+    msg: str = None,
+    detection_mode: bool = False,
+):
+    """
+    Get a standardized response object with patching instructions for the
+    request and error message.
+
+    Parameters
+    ----------
+    uid : str
+        The uid of the request, that was send to the Admission Controller.
+    allowed : bool
+        The decision, whether the request will be accepted or denied.
+    patch : list (optional)
+        A list with JSON patch instruction, that will modify the original
+        request, send to the Admission Controller. The list is Base64
+        encoded.
+    msg : str (optional)
+        The error message, which will be displayed, should allowed be
+        'False'.
+
+    Return
+    ----------
+    AdmissionReview : dict
+        Response is an AdmissionReview with following structure:
+
+        {
+          "apiVersion": "admission.k8s.io/v1beta1",
+          "kind": "AdmissionReview",
+          "response": {
+            "uid": uid,
+            "allowed": allowed,
+            "status": {
+                "code": 200,
+                "message": "All gucci, my boi."
+            },
+            "warnings": ["detection_mode ON"]
+            "patchType": "JSONPatch",
+            "patch":
+                "W3sib3AiOiAiYWRkIiwgInBhdGgiOiAiL3NwZWMvcmVwbGljYXMiLCAidmFsdWUiOiAzfV0="
+          }
+        }
+    """
+    review = {
+        "apiVersion": "admission.k8s.io/v1beta1",
```

Comment:
When migrating the webhook to `v1` API group one can decide which [admissionreviews the webhook should accept depending on the API group](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#request). I don't know whether one has to respond in the same group, but if so, we should make that configurable.

Reply:
But that can probably better be done by the one migrating the webhook then.

Reply:
#153